### ISSUE TITLE
[webview_flutter] How to avoid origin errors because it is "null" with iframes?

### ISSUE THREAD

**Issue by sezaru** [flutter/flutter#52367]
Hello, I'm trying to use TradingView Charting library with Flutter, as a reference, I'm using this Android implementation https://github.com/tradingview/charting-library-examples/blob/master/android/app/src/main/java/com/tradingview/android/MainActivity.kt

This is how I load it:

```
  @override
  Widget build(BuildContext context) => WebView(
    initialUrl: "about:blank",
    javascriptMode: JavascriptMode.unrestricted,
    onWebViewCreated: (controller) async {
      _controller = controller;

      var headers = <String, String>{
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'POST, GET, OPTIONS',
        'Access-Control-Allow-Credentials': 'true'
      };

      await _controller.loadUrl("file:///android_asset/flutter_assets/assets/charting_library/index.html", headers: headers);
    }
  );
```

It does work until the library creates an iframe (loading another file from the assets) and tries to change it via javascript, I will get a lot of error messages like this one:

`I/chromium(13460): [INFO:CONSOLE(1)] "Uncaught (in promise) SecurityError: Blocked a frame with origin "null" from accessing a frame with origin "null". Protocols, domains, and ports must match.", source: file:///android_asset/flutter_assets/assets/charting_library/charting_library/charting_library.min.js (1)`

It seems that the issue is that the origin is "null", is there some way to change it? I tried with the headers from the above code without success.

I'm not sure how to fix this and why it happens in flutter but not with the android example.

PS: I'm using latest flutter webview version ^0.9.6+2

<details>
  <summary>Logs</summary>

```
sezdocs@localhost ~/Downloads/traiding_view_test_app $ ~/flutter/bin/flutter run --verbose
[  +16 ms] executing: [/home/sezdocs/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +18 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 60d0fef17ae60b88cb52e6158bda2d91b600299c
[        ] executing: [/home/sezdocs/flutter/] git describe --match v*.*.* --first-parent --long --tags
[   +6 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.15.18-33-g60d0fef17
[   +5 ms] executing: [/home/sezdocs/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +2 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/home/sezdocs/flutter/] git ls-remote --get-url origin
[   +2 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +35 ms] executing: [/home/sezdocs/flutter/] git rev-parse --abbrev-ref HEAD
[   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[  +20 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +13 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb devices -l
[   +3 ms] Exit code 0 from: /home/sezdocs/Android/Sdk/platform-tools/adb devices -l
[        ] List of devices attached
           emulator-5554          device product:sdk_gphone_x86 model:sdk_gphone_x86 device:generic_x86_arm transport_id:2
[  +13 ms] /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell getprop
[  +33 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +75 ms] Found plugin webview_flutter at /home/sezdocs/.pub-cache/hosted/pub.dartlang.org/webview_flutter-0.3.19+9/
[  +40 ms] Found plugin webview_flutter at /home/sezdocs/.pub-cache/hosted/pub.dartlang.org/webview_flutter-0.3.19+9/
[  +33 ms] Generating /home/sezdocs/Downloads/traiding_view_test_app/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +17 ms] ro.hardware = ranchu
[  +12 ms] Using hardware rendering with device sdk gphone x86. If you get graphics artifacts, consider enabling software rendering with
           "--enable-software-rendering".
[  +17 ms] Launching lib/main.dart on sdk gphone x86 in debug mode...
[   +6 ms] /home/sezdocs/flutter/bin/cache/dart-sdk/bin/dart
/home/sezdocs/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root
/home/sezdocs/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter
-Ddart.developer.causal_async_stacks=true --output-dill /tmp/flutter_tool.OVKKTZ/app.dill --packages
/home/sezdocs/Downloads/traiding_view_test_app/.packages -Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions
--enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root
[   +7 ms] executing: /home/sezdocs/Android/Sdk/build-tools/29.0.3/aapt dump xmltree
/home/sezdocs/Downloads/traiding_view_test_app/build/app/outputs/apk/app.apk AndroidManifest.xml
[   +3 ms] Exit code 0 from: /home/sezdocs/Android/Sdk/build-tools/29.0.3/aapt dump xmltree
/home/sezdocs/Downloads/traiding_view_test_app/build/app/outputs/apk/app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1c
               A: android:compileSdkVersionCodename(0x01010573)="9" (Raw: "9")
               A: package="com.example.traidingviewtestapp" (Raw: "com.example.traidingviewtestapp")
               A: platformBuildVersionCode=(type 0x10)0x1c
               A: platformBuildVersionName=(type 0x10)0x9
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x11
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1c
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: uses-permission (line=15)
                 A: android:name(0x01010003)="android.permission.WAKE_LOCK" (Raw: "android.permission.WAKE_LOCK")
               E: uses-permission (line=16)
                 A: android:name(0x01010003)="android.permission.BLUETOOTH" (Raw: "android.permission.BLUETOOTH")
               E: application (line=24)
                 A: android:label(0x01010001)="traidingviewtestapp" (Raw: "traidingviewtestapp")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw:
                 "androidx.core.app.CoreComponentFactory")
                 E: activity (line=30)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.traidingviewtestapp.MainActivity" (Raw: "com.example.traidingviewtestapp.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=44)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=54)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw:
                     "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=58)
                     E: action (line=59)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=61)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=68)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +6 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell -x logcat -v time -t 1
[  +15 ms] Exit code 0 from: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell -x logcat -v time -t 1
[        ] --------- beginning of main
           03-10 19:49:37.621 I/Finsky  ( 8181): [524] jxw.run(3): Stats for Executor: bgExecutor jzt@d442d42[Running, pool size = 4, active
           threads = 0, queued tasks = 0, completed tasks = 2561]
[   +6 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb version
[        ] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 logcat -v time -T 03-10 19:49:37.621
[   +6 ms] <- compile package:traidingviewtestapp/main.dart
[   +6 ms] Android Debug Bridge version 1.0.41
           Version 29.0.6-6198805
           Installed as /home/sezdocs/Android/Sdk/platform-tools/adb
[   +1 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb start-server
[   +3 ms] Building APK
[  +15 ms] Running Gradle task 'assembleDebug'...
[   +1 ms] gradle.properties already sets `android.enableR8`
[   +1 ms] Using gradle from /home/sezdocs/Downloads/traiding_view_test_app/android/gradlew.
[        ] /home/sezdocs/Downloads/traiding_view_test_app/android/gradlew mode: 33261 rwxr-xr-x.
[   +9 ms] executing: /opt/android-studio/jre/bin/java -version
[  +53 ms] Exit code 0 from: /opt/android-studio/jre/bin/java -version
[        ] openjdk version "1.8.0_232"
           OpenJDK Runtime Environment (IcedTea 3.14.0) (Gentoo icedtea-3.14.0)
           OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)
[   +3 ms] executing: [/home/sezdocs/Downloads/traiding_view_test_app/android/] /home/sezdocs/Downloads/traiding_view_test_app/android/gradlew
-Pverbose=true -Ptarget-platform=android-x86 -Ptarget=/home/sezdocs/Downloads/traiding_view_test_app/lib/main.dart -Ptrack-widget-creation=true
-Pfilesystem-scheme=org-dartlang-root assembleDebug
[ +586 ms] > Task :app:compileFlutterBuildDebug UP-TO-DATE
[        ] > Task :app:packLibsflutterBuildDebug UP-TO-DATE
[        ] > Task :app:preBuild UP-TO-DATE
[        ] > Task :app:preDebugBuild UP-TO-DATE
[        ] > Task :webview_flutter:preBuild UP-TO-DATE
[        ] > Task :webview_flutter:preDebugBuild UP-TO-DATE
[        ] > Task :webview_flutter:packageDebugRenderscript NO-SOURCE



> Task :app:compileDebugRenderscript NO-SOURCE
[        ] > Task :app:generateDebugBuildConfig UP-TO-DATE
[        ] > Task :app:cleanMergeDebugAssetsbugAidl NO-SOURCE
[        ] > Task :app:mergeDebugShaders UP-TO-DATE
[        ] > Task :app:compileDebugShaders UP-TO-DATE
[        ] > Task :app:generateDebugAssets UP-TO-DATE
[        ] > Task :webview_flutter:mergeDebugShaders UP-TO-DATE
[        ] > Task :webview_flutter:compileDebugShaders UP-TO-DATE
[        ] > Task :webview_flutter:generateDebugAssets UP-TO-DATE
[        ] > Task :webview_flutter:packageDebugAssets UP-TO-DATE
[        ] > Task :app:mergeDebugAssets
[ +180 ms] > Task :app:copyFlutterAssetsDebug
[        ] > Task :app:mainApkListPersistenceDebug UP-TO-DATE
[        ] > Task :app:generateDebugResValues UP-TO-DATE
[        ] > Task :app:generateDebugResources UP-TO-DATE
[        ] > Task :webview_flutter:generateDebugResValues UP-TO-DATE
[        ] > Task :webview_flutter:compileDebugRenderscript NO-SOURCE
[        ] > Task :webview_flutter:generateDebugResources UP-TO-DATE
[        ] > Task :webview_flutter:packageDebugResources UP-TO-DATE
[        ] > Task :app:mergeDebugResources UP-TO-DATE
[        ] > Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
[        ] > Task :webview_flutter:checkDebugManifest UP-TO-DATE
[        ] > Task :webview_flutter:processDebugManifest UP-TO-DATE
[        ] > Task :app:processDebugManifest UP-TO-DATE
[        ] > Task :webview_flutter:parseDebugLibraryResources UP-TO-DATE
[        ] > Task :webview_flutter:generateDebugRFile UP-TO-DATE
[        ] > Task :app:processDebugResources UP-TO-DATE
[        ] > Task :webview_flutter:generateDebugBuildConfig UP-TO-DATE
[        ] > Task :app:processDebugJavaRes NO-SOURCE
[        ] > Task :webview_flutter:javaPreCompileDebug UP-TO-DATE
[        ] > Task :webview_flutter:compileDebugJavaWithJavac UP-TO-DATE
[        ] > Task :webview_flutter:bundleLibCompileDebug UP-TO-DATE
[        ] > Task :app:compileDebugKotlin UP-TO-DATE
[        ] > Task :app:javaPreCompileDebug UP-TO-DATE
[        ] > Task :app:compileDebugJavaWithJavac UP-TO-DATE
[        ] > Task :app:compileDebugSources UP-TO-DATE
[        ] > Task :webview_flutter:processDebugJavaRes NO-SOURCE
[        ] > Task :webview_flutter:bundleLibResDebug UP-TO-DATE
[        ] > Task :app:mergeDebugJavaResource UP-TO-DATE
[        ] > Task :webview_flutter:bundleLibRuntimeDebug UP-TO-DATE
[        ] > Task :webview_flutter:createFullJarDebug UP-TO-DATE
[        ] > Task :app:checkDebugDuplicateClasses UP-TO-DATE
[        ] > Task :app:desugarDebugFileDependencies UP-TO-DATE
[  +22 ms] > Task :app:transformClassesWithDexBuilderForDebug UP-TO-DATE
[        ] > Task :app:mergeExtDexDebug UP-TO-DATE
[        ] > Task :app:mergeDexDebug UP-TO-DATE
[        ] > Task :app:validateSigningDebug UP-TO-DATE
[        ] > Task :app:signingConfigWriterDebug UP-TO-DATE
[        ] > Task :app:mergeDebugJniLibFolders UP-TO-DATE
[        ] > Task :webview_flutter:mergeDebugJniLibFolders UP-TO-DATE
[        ] > Task :webview_flutter:mergeDebugNativeLibs UP-TO-DATE
[        ] > Task :webview_flutter:stripDebugDebugSymbols UP-TO-DATE
[        ] > Task :webview_flutter:transformNativeLibsWithIntermediateJniLibsForDebug UP-TO-DATE
[        ] > Task :app:mergeDebugNativeLibs UP-TO-DATE
[        ] > Task :app:stripDebugDebugSymbols UP-TO-DATE
[        ] Compatible side by side NDK version was not found.
[        ] > Task :app:packageDebug UP-TO-DATE
[        ] > Task :app:assembleDebug UP-TO-DATE
[        ] > Task :webview_flutter:extractDebugAnnotations UP-TO-DATE
[        ] > Task :webview_flutter:mergeDebugGeneratedProguardFiles UP-TO-DATE
[        ] > Task :webview_flutter:mergeDebugConsumerProguardFiles UP-TO-DATE
[        ] > Task :webview_flutter:prepareLintJarForPublish UP-TO-DATE
[        ] > Task :webview_flutter:mergeDebugJavaResource UP-TO-DATE
[        ] > Task :webview_flutter:transformClassesAndResourcesWithSyncLibJarsForDebug UP-TO-DATE
[        ] > Task :webview_flutter:transformNativeLibsWithSyncJniLibsForDebug UP-TO-DATE
[        ] > Task :webview_flutter:bundleDebugAar UP-TO-DATE
[        ] > Task :webview_flutter:compileDebugSources UP-TO-DATE
[        ] > Task :webview_flutter:assembleDebug UP-TO-DATE
[        ] BUILD SUCCESSFUL in 733ms
[        ] 58 actionable tasks: 3 executed, 55 up-to-date
[ +324 ms] Running Gradle task 'assembleDebug'... (completed in 1.2s)
[  +16 ms] calculateSha: LocalDirectory: '/home/sezdocs/Downloads/traiding_view_test_app/build/app/outputs/apk'/app.apk
[  +22 ms] calculateSha: reading file took 22us
[ +441 ms] calculateSha: computing sha took 440us
[   +3 ms] ✓ Built build/app/outputs/apk/debug/app-debug.apk.
[   +7 ms] executing: /home/sezdocs/Android/Sdk/build-tools/29.0.3/aapt dump xmltree
/home/sezdocs/Downloads/traiding_view_test_app/build/app/outputs/apk/app.apk AndroidManifest.xml
[   +4 ms] Exit code 0 from: /home/sezdocs/Android/Sdk/build-tools/29.0.3/aapt dump xmltree
/home/sezdocs/Downloads/traiding_view_test_app/build/app/outputs/apk/app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1c
               A: android:compileSdkVersionCodename(0x01010573)="9" (Raw: "9")
               A: package="com.example.traidingviewtestapp" (Raw: "com.example.traidingviewtestapp")
               A: platformBuildVersionCode=(type 0x10)0x1c
               A: platformBuildVersionName=(type 0x10)0x9
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x11
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1c
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: uses-permission (line=15)
                 A: android:name(0x01010003)="android.permission.WAKE_LOCK" (Raw: "android.permission.WAKE_LOCK")
               E: uses-permission (line=16)
                 A: android:name(0x01010003)="android.permission.BLUETOOTH" (Raw: "android.permission.BLUETOOTH")
               E: application (line=24)
                 A: android:label(0x01010001)="traidingviewtestapp" (Raw: "traidingviewtestapp")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw:
                 "androidx.core.app.CoreComponentFactory")
                 E: activity (line=30)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.traidingviewtestapp.MainActivity" (Raw: "com.example.traidingviewtestapp.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=44)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=54)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw:
                     "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=58)
                     E: action (line=59)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=61)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=68)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +1 ms] Stopping app 'app.apk' on sdk gphone x86.
[        ] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell am force-stop com.example.traidingviewtestapp
[  +94 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell pm list packages com.example.traidingviewtestapp
[  +34 ms] package:com.example.traidingviewtestapp
[   +2 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell cat
/data/local/tmp/sky.com.example.traidingviewtestapp.sha1
[  +50 ms] a46d0ee522ce6e1a3d86a27a652fc94f237e50cb
[        ] Latest build already installed.
[        ] sdk gphone x86 startApp
[   +2 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 shell am start -a android.intent.action.RUN -f 0x20000000 --ez
enable-background-compilation true --ez enable-dart-profiling true --ez enable-checked-mode true --ez verify-entry-points true
com.example.traidingviewtestapp/com.example.traidingviewtestapp.MainActivity
[ +111 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=com.example.traidingviewtestapp/.MainActivity (has extras) }
[        ] Waiting for observatory port to be available...
[ +584 ms] Observatory URL on device: http://127.0.0.1:38839/goPCn3vOTrE=/
[   +1 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 forward tcp:0 tcp:38839
[   +4 ms] 33349
[        ] Forwarded host port 33349 to device port 38839 for Observatory
[+2237 ms] Connecting to service protocol: http://127.0.0.1:33349/goPCn3vOTrE=/
[ +142 ms] Successfully connected to service protocol: http://127.0.0.1:33349/goPCn3vOTrE=/
[   +2 ms] Sending to VM service: getVM({})
[   +3 ms] Result: {type: VM, name: vm, architectureBits: 32, hostCPU: Android virtual processor, operatingSystem: android, targetCPU: ia32,
version: 2.8.0-dev.12.0.flutter-9983424a3c (Thu Mar 5 16:51:44 2020 +0000) on "android_ia32", _profilerMode: VM, _nativeZoneM...
[   +4 ms] Sending to VM service: getIsolate({isolateId: isolates/1255608627748839})
[   +3 ms] Sending to VM service: _flutter.listViews({})
[  +20 ms] Result: {type: Isolate, id: isolates/1255608627748839, name: main, number: 1255608627748839, _originNumber: 1255608627748839, startTime:
1583880588312, _heaps: {new: {type: HeapSpace, name: new, vmName: Scavenger, collections: 3, avgCollectionPeriodMillis...
[  +12 ms] Result: {type: FlutterViewList, views: [{type: FlutterView, id: _flutterView/0xf474bb10, isolate: {type: @Isolate, fixedId: true, id:
isolates/1255608627748839, name: main.dart$main-1255608627748839, number: 1255608627748839}}]}
[   +3 ms] DevFS: Creating new filesystem on the device (null)
[        ] Sending to VM service: _createDevFS({fsName: traiding_view_test_app})
[  +20 ms] Result: {type: FileSystem, name: traiding_view_test_app, uri:
file:///data/user/0/com.example.traidingviewtestapp/code_cache/traiding_view_test_appGCBVHQ/traiding_view_test_app/}
[        ] DevFS: Created new filesystem on the device
(file:///data/user/0/com.example.traidingviewtestapp/code_cache/traiding_view_test_appGCBVHQ/traiding_view_test_app/)
[   +2 ms] Updating assets
[ +333 ms] Syncing files to device sdk gphone x86...
[   +1 ms] Scanning asset files
[   +8 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[        ] <- recompile package:traidingviewtestapp/main.dart 207ced1e-7a7b-4737-87d5-19a5f1681010
[        ] <- 207ced1e-7a7b-4737-87d5-19a5f1681010
[  +47 ms] Updating files
[  +85 ms] I/chatty  (13818): uid=10149(com.example.traidingviewtestapp) RenderThread identical 1 line
[        ] D/EGL_emulation(13818): eglMakeCurrent: 0xdc31a0c0: ver 2 0 (tinfo 0xdc30f3f0)
[+1046 ms] DevFS: Sync finished
[   +1 ms] Syncing files to device sdk gphone x86... (completed in 1,190ms)
[        ] Synced 7.5MB.
[   +1 ms] Sending to VM service: _flutter.listViews({})
[   +1 ms] Result: {type: FlutterViewList, views: [{type: FlutterView, id: _flutterView/0xf474bb10, isolate: {type: @Isolate, fixedId: true, id:
isolates/1255608627748839, name: main.dart$main-1255608627748839, number: 1255608627748839}}]}
[        ] <- accept
[        ] Connected to _flutterView/0xf474bb10.
[   +1 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on sdk gphone x86 is available at: http://127.0.0.1:33349/goPCn3vOTrE=/
[+3703 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +1 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 forward --list
[   +4 ms] Exit code 0 from: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 forward --list
[   +1 ms] DevFS: Deleting filesystem on the device
(file:///data/user/0/com.example.traidingviewtestapp/code_cache/traiding_view_test_appGCBVHQ/traiding_view_test_app/)
[        ] Sending to VM service: _deleteDevFS({fsName: traiding_view_test_app})
[ +251 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +1 ms] executing: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 forward --list
[   +3 ms] Exit code 0 from: /home/sezdocs/Android/Sdk/platform-tools/adb -s emulator-5554 forward --list
[   +1 ms] "flutter run" took 10,921ms.
```

```
sezdocs@localhost ~/Downloads/traiding_view_test_app $ ~/flutter/bin/flutter analyze
Analyzing traiding_view_test_app...                                     

   info • Unused import: 'dart:convert' • lib/webview_demo.dart:2:8 • unused_import
   info • Unused import: 'package:flutter/services.dart' • lib/webview_demo.dart:5:8 • unused_import
   info • Unused import: 'package:traidingviewtestapp/chart_widget.dart' • lib/webview_demo.dart:7:8 • unused_import
   info • Unused import: 'chat_lib.dart' • lib/webview_demo.dart:11:8 • unused_import

4 issues found. (ran in 2.4s)
```

```
sezdocs@localhost ~/Downloads/traiding_view_test_app $ ~/flutter/bin/flutter doctor -v
[✓] Flutter (Channel master, v1.15.19, on Linux, locale en_US.utf8)
    • Flutter version 1.15.19 at /home/sezdocs/flutter
    • Framework revision 60d0fef17a (18 hours ago), 2020-03-09 22:07:34 -0700
    • Engine revision 0e2c8feeeb
    • Dart version 2.8.0 (build 2.8.0-dev.12.0 9983424a3c)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/sezdocs/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (IcedTea 3.14.0) (Gentoo icedtea-3.14.0)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup for detailed instructions.

[✗] Chrome - develop for the web (Cannot find chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang++ 9.0.1
    • GNU Make 4.3

[✓] Android Studio (version 3.6)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (IcedTea 3.14.0) (Gentoo icedtea-3.14.0)

[✓] Connected device (2 available)
    • Linux      • Linux      • linux-x64      • Linux
    • Web Server • web-server • web-javascript • Flutter Tools

! Doctor found issues in 2 categories.
```

</details>

**Comment by tonyclickspace** [comment]
Not sure if this is the same issue or just related, but seeing a similar issue with two-way communication using postMessage and HTML files loaded from the filesystem.

If the HTML calls postMessage with a targetOrigin other than "*", an error is thrown:
`Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://example.com') does not match the recipient window's origin ('null').`
and if the origin is set to the string "null":
`Uncaught SyntaxError: Failed to execute 'postMessage' on 'Window': Invalid target origin 'null' in a call to 'postMessage'.`

**Comment by danagbemava-nc** [comment]
Hi @sezaru, can you confirm if you're still experiencing this issue on the latest `flutter` and `webview_flutter` versions?.

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by banitaparamanik** [comment]
Access to XMLHttpRequest at 'file:///android_asset/flutter_assets/assets/model/r2-d2.obj' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, https.", source: file:///android_asset/flutter_assets/assets/three.min.js (14669)

I am getting this type of error using webview_flutter during loading assets file
